### PR TITLE
Implemented Soap and Rest probing mock services

### PR DESCRIPTION
**Pull request by gdellorusso**
Implemented Soap and Rest probing mock services for testing interop-be-probing-operations in dev environment

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			<version>${javax.validation.api}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>servlet-api</artifactId>
+			<version>2.5</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml]: Use properties

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			<artifactId>lombok</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>wsdl4j</groupId>
+			<artifactId>wsdl4j</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>javax.xml.ws</groupId>
+			<artifactId>jaxws-api</artifactId>
+			<version>2.3.1</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml]: See above

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		</dependency>
+		<dependency>
+			<groupId>javax.jws</groupId>
+			<artifactId>javax.jws-api</artifactId>
+			<version>1.1</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml]: See above

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.glassfish.jaxb</groupId>
+			<artifactId>jaxb-runtime</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.sun.activation</groupId>
+			<artifactId>jakarta.activation</artifactId>
+			<version>1.2.1</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml]: See above

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>wsdl4j</groupId>
+			<artifactId>wsdl4j</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>javax.xml.ws</groupId>
```
Comment (nttdata-rtorsoli) [inline pom.xml:78]: Better to use CXF , it works better with Spring

**Review by nttdata-rtorsoli — DISMISSED** [review]
See comments

**Comment by gdellorusso** [comment]
since it is a mock-up that will later be deleted, the development was deliberately not done perfectly

**Comment by gdellorusso** [comment]
given that it is a mock-up that will later be deleted, the development was intentionally not done perfectly

File: src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+public class ProbingController {
+
+  @GetMapping(value = "/probing/v1", produces = "application/json")
+  public ResponseEntity probingOk() {
+    log.info("REST call with OK result ");
+
+    return ResponseEntity.ok().build();
+  }
+
+  @GetMapping(value = "/probingError/v1", produces = "application/json")
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java:22]: this is never called, right?

Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java:22]: yes, it is only for testing.
Furthermore, the REST endpoints have the /rest prefix as they would conflict with the "/probing" SOAP.

Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java:22]: we can use it by setting it as a basepath from db to test the error case

**Review by nttdata-rtorsoli — APPROVED** [review]

File: src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java (hunk 6 above)
Comment (galales) [reply src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java:22]: a nice to have feature could be to return an error 10 or 20% of the time, so we do not need to switch back and forth to test the error case

**Review by galales — APPROVED** [review]

File: src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java (hunk 6 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/eservice/mock/controller/ProbingController.java:22]: Ok, if we need it, we could include it as an improvement task

**Pull request merged**